cms/djangoapps/contentstore/views/block.py

```diff
@@ -300,6 +300,118 @@ def xblock_view_handler(request, usage_key_string, view_name):
+        raise PermissionDenied()
+
+    store = modulestore()
+    xblock = store.get_item(usage_key)
+
+    # wrap the generated fragment in the xmodule_editor div so that the javascript
+    # can bind to it correctly
+    xblock.runtime.wrappers.append(
+        partial(
+            wrap_xblock,
+            "StudioRuntime",
+            usage_id_serializer=str,
+            request_token=request_token(request),
+        )
+    )
+
+    xblock.runtime.wrappers_asides.append(
+        partial(
+            wrap_xblock_aside,
+            "StudioRuntime",
+            usage_id_serializer=str,
+            request_token=request_token(request),
+            extra_classes=["wrapper-comp-plugins"],
+        )
+    )
+
+    load_services_for_studio(xblock.runtime, request.user)
+
+    # try:
+    studio_fragment = xblock.render("studio_view")
+    # catch exceptions indiscriminately, since after this point they escape the
+    # dungeon and surface as uneditable, unsaveable, and undeletable
+    # component-goblins.
+    # except Exception as exc:  # pylint: disable=broad-except
+    #     log.debug(
+    #         "Unable to render %s for %r", "STUDIO_VIEW", xblock, exc_info=True
+    #     )
+    #     fragment = Fragment(
+    #         render_to_string("html_error.html", {"message": str(exc)})
+    #     )
+
+    can_edit = has_studio_write_access(request.user, usage_key.course_key)
+
+    # Determine the items to be shown as reorderable. Note that the view
+    # 'reorderable_container_child_preview' is only rendered for xblocks that
+    # are being shown in a reorderable container, so the xblock is automatically
+    # added to the list.
+    # reorderable_items = set()
+
+    force_render = request.GET.get("force_render", None)
+
+    # Fetch tags of children components
+    tags_count_map = {}
+    if use_tagging_taxonomy_list_page():
+        tags_count_map = get_children_tags_count(xblock)
+
+    # Set up the context to be passed to each XBlock's render method.
+    context = request.GET.dict()
+    context.update(
+        {
+            # This setting disables the recursive wrapping of xblocks
+            "is_pages_view": True,
+            "is_unit_page": is_unit(xblock),
+            "can_edit": can_edit,
+            "root_xblock": None,
+            "reorderable_items": None,
+            "paging": None,
+            "force_render": force_render,
+            "item_url": "/container/{usage_key}",
+            "tags_count_map": tags_count_map,
+        }
+    )
+    wrapper_fragment = get_preview_fragment(request, xblock, context)
+
+    # Note that the container view recursively adds headers into the preview fragment,
+    # so only the "Pages" view requires that this extra wrapper be included.
+    display_label = xblock.display_name or xblock.scope_ids.block_type
+    if not xblock.display_name and xblock.scope_ids.block_type == "html":
+        display_label = _("Text")
+    wrapper_fragment.content = render_to_string(
```

Comment:
Eventually this variable isn't used anywhere, let's remove this block. Seems like `get_container_handler_context` returns all required context